cmake/openenclave-config.cmake.in

```diff
@@ -90,20 +90,24 @@ if (NOT TARGET openenclave::sgx_enclave_common)
   endif ()
 endif ()
 
-if (NOT TARGET openenclave::sgx_dcap_ql)
-  find_library(SGX_DCAP_QL_LIB NAMES sgx_dcap_ql
-               HINTS ${NUGET_PACKAGE_PATH}/DCAP_Components/build/lib/native/Libraries)
-  if (NOT SGX_DCAP_QL_LIB)
-    message(FATAL_ERROR "-- Looking for sgx_dcap_ql library - not found")
-  else ()
-    message(VERBOSE "-- Looking for sgx_dcap_ql library - found")
-    add_library(openenclave::sgx_dcap_ql SHARED IMPORTED)
-    if (UNIX)
-      set_target_properties(openenclave::sgx_dcap_ql PROPERTIES IMPORTED_LOCATION ${SGX_DCAP_QL_LIB})
-    elseif (WIN32)
-      set_target_properties(openenclave::sgx_dcap_ql PROPERTIES
-                            IMPORTED_LOCATION $ENV{WINDIR}/System32
-                            IMPORTED_IMPLIB ${SGX_DCAP_QL_LIB})
+if (WITHOUT_DCAP_LINK)
```

Comment:
Does this work?

- If the OE SDK is compiled against `sgx_dcap_ql`, it will have a missing symbol at link time if the required library is not provided via `target_link_libraries`. This is the only configuration that the OE SDK currently ships with.

- If the OE SDK is not compiled against `sgx_dcap_ql`, then it shouldn't require a manual `WITHOUT_DCAP_LINK`; the inclusion can opportunistically include the library iff it is found. (i.e. failure to find a library results in a warning, which will result in a link time rather than cmake time failure; finding a library results in adding it as a link target which is linked or not as needed depending on how liboehost was compiled.

Reply:
It works for the case we're talking about here:
On Windows, when you build the OE SDK with the USE_LIBSGX=OFF flag, and want to run the samples successfully, and will want to link with libsgx_enclave_common with the upcoming changes that @jxyang is working on.

The cmake package config doesn't have any tracking variables for whether or not it was built with USE_LIBSGX=ON or OFF, because we aren't going to be publishing any packages for any platform with USE_LIBSGX=OFF. However, if a user really wants to build applications with the OE SDK with USE_LIBSGX=OFF and they're willing to build the OE SDK from source and then include/link against it, then the implementation provided in this PR will allow you to do that if you specify the WITHOUT_DCAP_LINK=ON flag.

For the two cases you talk about here:
1. `If the OE SDK is compiled against sgx_dcap_ql`, then they should not set this flag or they won't be able to link their host applications without symbols missing, like you mentioned.
2. `If the OE SDK is not compiled against sgx_dcap_ql`, then this is the case where they built the OE SDK with USE_LIBSGX=OFF and will want to set this flag to on.

So this flag very much related to whether or not the SDK was built with the USE_LIBSGX option. We could simplify it by simply tracking the USE_LIBSGX in the cmake package, but I feel like the naming for that flag is *REALLY* bad for this situation, because people will still need to link against libsgx_enclave_common on Windows with the upcoming enclave launching changes.

I'm not sure we should be linking with the library iff it is found and announce a warning if it isn't found. In my experience, people almost *always* ignore warnings, and usually ignore errors too, unless the errors are blocking them :) But I'm fine with adjusting this. Really, I'm just trying to help unblock @jxyang 's work, feel free to take these changes into your PR and adjust them however you wish!

Reply:
Thanks @johnkord. I took some of your changes in this file, but renamed the definition to `HAS_QUOTE_PROVIDER` (I feel any name referring to dcap is too technical and subject to change soon). When this definition is off, OE SDK and samples link with enclave common api library, but not dcap library. When it is on, both libraries are linked. OE SDK and samples are always in sync w.r.t. this definition. In other words, for folks on a non-FLC machine, they have to rebuilt OE SDK to run the samples.

Reply:
@jxyang Sounds good! That's a good name for this variable too! I'll close this PR and leave you to include those changes in your PR!